### ISSUE TITLE
Distraction free mode layout is broken

### ISSUE THREAD

**Issue by C0deboy** [vsch/idea-multimarkdown#593]
It's me again... I found that if I toggle distraction free mode (ctrl+alt+shift+d)  then layout is not adjusted. I use this view to write blogs so this is quite must-have feature. In standard markdown support it works perfectly. Text is centered and margins are big.

Using markdown navigator text is not wrapped, it even goes off the screen and scroll X is visible. I'm not sure if it's case of your plugin or IDE settings but with standard markdown support it works as expected. 

Edit: It seems like right margin is not respected and text is not wrapped.

**Comment by C0deboy** [comment]
Yes, I reviewed all the options before but I can't get it working like in standard markdown support. If I tick "soft wrap on right margin" it's working but it is GLOBAL. It applies it to normal editing too whereas when using standard markdown support text is only wrapped in distraction free mode.

**Comment by vsch** [comment]
@C0deboy, I am trying out Markdown Support in distraction free mode and I am not getting any wrapping at the right margin unless soft-wrap is on.

@C0deboy, basically what you are expecting is that soft-wrap at right margin is enabled in distraction free mode regardless of the setting in options. However, soft-wraps will need to be enabled for this to work.

Is my understanding correct?

**Comment by C0deboy** [comment]
Let me do some screenshots for you so you can understand me 100%.

Here are screenshots representing how it works in markdown support:

Normal mode:
![capture](https://user-images.githubusercontent.com/22492109/39777308-8054180a-5303-11e8-9bf8-3b109be75064.PNG)
Distraction free:
![capture2](https://user-images.githubusercontent.com/22492109/39777328-8ce50c14-5303-11e8-9e50-1b8cbc393e35.PNG)

And that is how markdown navigator behaves:

Normal mode:
![capture3](https://user-images.githubusercontent.com/22492109/39777343-9f1b7a4e-5303-11e8-9ee3-a58f7c3d398c.PNG)
Distraction free:
![capture4](https://user-images.githubusercontent.com/22492109/39777356-a7440cfe-5303-11e8-889e-3bc90bc69b2a.PNG)

Basically nothing changes except that tool windows are closed.

**Comment by vsch** [comment]
@C0deboy, this is what I described. Markdown Support has soft-wraps at right margin enabled in distraction free mode. If you turn off soft-wraps then in distraction free mode, Markdown Support will also have nothing happening.

Markdown Support does not have wrapping at right margin nor separate soft-wraps mode for markdown documents so the default IDE behaviour works in distraction free mode.

Markdown Navigator adds special handling to allow having both features available separate from the global IDE setting and allow soft-wrapping at right margin without needing to be in distraction free mode. The code handling these features overrides the default IDE behaviour in distraction free mode.

I will add the logic to force soft-wrap on right-margin in distraction free mode regardless of the setting in options.

Why do you only want soft-wrapping at right margin in distraction free mode?

Markdown is text and it would seem that having it wrapped to right margin makes for easier reading whether you are in distraction free mode or not. This was a user request and it makes sense since viewing text is easier if it is not spanning the full width of the window.

Myself, I don't use either mode and always have my markdown documents hard-wrapped to margins using markdown document format or markdown element format. 

That is why it is not clear to me what the advantage is in markdown documents not to soft-wrap to right margin, if soft-wraps is something that you use. The only use cases are for dealing with wide tables, in which case nothing makes it easier to read them and the other is for editing comments which treat line breaks as hard wraps on GitHub and other forums.

**Comment by C0deboy** [comment]
It is just something that I get used to, but it's ok. Maybe it's even better ;) Important thing is it's working in distraction free mode, so it resolves my issue.  Thanks for your assistance.
